### ISSUE TITLE
Finish carving out geometry and gradient from `drake/util` into new homes

### ISSUE THREAD

**Issue by jwnimmer-tri** [RobotLocomotion/drake#3414]
The goal is for these include statements to go away, replaced by paths into `common` or `math` (or removed) as necessary:

```
./systems/plants/constraint/RigidBodyConstraint.cpp:11:#include "drake/util/drakeGeometryUtil.h"
./systems/plants/joints/FixedAxisOneDoFJoint.h:12:#include "drake/util/drakeGradientUtil.h"
./systems/plants/joints/QuaternionFloatingJoint.h:7:#include "drake/util/drakeGeometryUtil.h"
./systems/plants/joints/RollPitchYawFloatingJoint.h:7:#include "drake/util/drakeGeometryUtil.h"
./systems/plants/RigidBody.cpp:5:#include "drake/util/drakeGeometryUtil.h"
./systems/plants/RigidBodyTree.cpp:13:#include "drake/util/drakeGeometryUtil.h"
./systems/plants/RigidBodyTree.h:26:#include "drake/util/drakeGeometryUtil.h"
./systems/plants/test/testKinematicsCacheChecks.cpp:11:#include "drake/util/drakeGeometryUtil.h"
./systems/robotInterfaces/QPLocomotionPlan.cpp:19:#include "drake/util/drakeGeometryUtil.h"
```

This may involved relocating the helper functions.

This is a portion of #3278 and thus a prereq for #3253.

**Comment by jwnimmer-tri** [comment]
/CC @david-german-tri and @hongkai-dai perhaps one of you could take this one over for us?

**Comment by david-german-tri** [comment]
This parallelizes well.  I'm happy to work on chunks of it in the background.  Before starting to work on a piece of it, I'll post here; could others please do the same?

**Comment by hongkai-dai** [comment]
I can handle the drakeGeometryUtil include

**Comment by jwnimmer-tri** [comment]
Was this close intentional?  There's a bunch of non-forwarding-methods still left in drakeGeometryUtil.  Should we just delete them?

**Comment by hongkai-dai** [comment]
@jwnimmer-tri you are right. I thought many of these functions can be replaced by using AutoDiffScalar, but we have not implemented the template version with AutoDiffScalar yet.

**Comment by david-german-tri** [comment]
@hongkai-dai, I'm unassigning myself.

**Comment by jwnimmer-tri** [comment]
Fixed in #9935 -- util is in the attic now.
